pkg/generators/forwarding/fluentd/output_conf_syslog_test.go

```diff
@@ -0,0 +1,56 @@
+package fluentd
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+	test "github.com/openshift/cluster-logging-operator/test"
+)
+
+var _ = Describe("Generating external syslog server output store config blocks", func() {
+
+	var (
+		err       error
+		outputs   []logging.OutputSpec
+		generator *ConfigGenerator
+	)
+	BeforeEach(func() {
+		generator, err = NewConfigGenerator(false)
+		Expect(err).To(BeNil())
+	})
+
+	Context("for an insecure endpoint", func() {
+		BeforeEach(func() {
+			outputs = []logging.OutputSpec{
+				{
+					Type:     logging.OutputTypeSyslog,
+					Name:     "syslog-receiver",
+					Endpoint: "sl.svc.messaging.cluster.local:9654",
+				},
+			}
+		})
+		It("should produce well formed output label config", func() {
+			results, err := generator.generateOutputLabelBlocks(outputs)
+			Expect(err).To(BeNil())
+			Expect(len(results)).To(Equal(1))
+			test.Expect(results[0]).ToEqual(`<label @SYSLOG_RECEIVER>
+	<match **>
+	   @type copy
+	   <store>
+  	     @type "#{ENV['SYSLOG_TYPE'] || 'syslog_buffered'}"
+	     @id syslog_receiver
+	     remote_syslog sl.svc.messaging.cluster.local
+	     port 9654
+	     hostname ${hostname}
+	     facility "#{ENV['SYSLOG_FACILITY'] || 'user'}"
```

Comment:
@richm if we are using LogForwarding and requiring the only way to modify conf is to:

1. Set ClusterLogging(Logforwarding?) to Unmanaged
1. Manually edit conf

Do you see any reason to still rely on env vars in lieu of explicitly setting a value?  I realize we have this elsewhere in our config generation but seems pointless given the mechanism to modify it.

Reply:
@ahadas per my discussion with @ewolinetz please explicitly set these values in lieu of them optionally being set by envvar.  Given the the fact the config is generated and to change anything about the config or DaemonSet env vars require you to go unmanaged, it makes more sense to simply require someone to update config directly.  Aside from that it lgtm

Reply:
Done